File: Ballast.Core/src/Client/models-old/board-type.ts
export interface IBoardType {
    value: number;
    name: string;
}

export class BoardType implements IBoardType {

    public static Square: BoardType = new BoardType({ value: 0, name: 'Square' });
    public static Octagon: BoardType = new BoardType({ value: 1, name: 'Octagon' });
    
    public value!: number;
    public name!: string;

    private constructor(state: IBoardType) {
        this.setState(state);
    }

    private setState(state: IBoardType): BoardType {
        this.value = state.value;
        this.name = state.name;
        return this;
    }

    public static list(): BoardType[] {
        return [
            BoardType.Square,
            BoardType.Octagon
        ];
    }

    public static fromValue(value: number): BoardType {
        if (!!value) {
            switch(value) {
                case (BoardType.Square.value):
                    return BoardType.Square;
                case (BoardType.Octagon.value):
                    return BoardType.Octagon;
            }
        }
        throw new Error(`Could derive board type from value (${value})`);
    }

    public static fromString(text: string): BoardType {
        if (!!text) {
            switch(text.toLocaleLowerCase()) {
                case (BoardType.Square.name.toLocaleLowerCase()):
                    return BoardType.Square;
                case (BoardType.Octagon.name.toLocaleLowerCase()):
                    return BoardType.Octagon;
            }
        }
        throw new Error(`Could derive board type from text/name (${text})`);
    }

}

File: Ballast.Core/src/Server/ModelsOld/BoardType.cs
// using System;
// using System.Collections.Generic;
// using System.Linq;

// namespace Ballast.Core.Models
// {
//     public class BoardType
//     {
//         public static BoardType Square => new BoardType(0, "Square");
//         public static BoardType Octagon => new BoardType(1, "Octagon");

//         public int Value { get; private set; }
//         public string Name { get; private set; }

//         private BoardType(int value, string name)
//         {
//             Value = value;
//             Name = name;
//         }

//         public static IEnumerable<BoardType> List()
//         {
//             // alternately, use a dictionary keyed by value
//             return new[] {
//                 Square,
//                 Octagon
//             };
//         }

//         public static BoardType FromString(string text)
//         {
//             return List().Single(x => String.Equals(x.Name, text, StringComparison.OrdinalIgnoreCase));
//         }

//         public static BoardType FromValue(int value)
//         {
//             return List().Single(x => x.Value == value);
//         }
//     }
// }
